languages/zh-tw/constants.genie

```diff
@@ -0,0 +1,134 @@
+    // rule if the flag is set
+
+    !turking {
+    '0'    => new Ast.Value.Number(0);
+    '零' => new Ast.Value.Number(0);
+    '1'    => new Ast.Value.Number(1);
+    '一'  => new Ast.Value.Number(1);
+    }
+}
+
+constant_date_point = {
+    '現在'             => new Ast.Value.Date(null, '+', null);
+    '今天'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'day'), '+', null);
+    '昨天'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'day'), '-', Ast.Value.Measure(1, 'day'));
+    '這 星期'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '+', null);
+    '這週'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '+', null);
+    '這 禮拜'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '+', null);
+
+    !turking {
+    '明天'            => new Ast.Value.Date(Ast.DateEdge('end_of', 'day'), '+', null);
+    '今天 以前'         => new Ast.Value.Date(Ast.DateEdge('end_of', 'day'), '+', null);
+    '這 星期 以前'       => new Ast.Value.Date(Ast.DateEdge('end_of', 'week'), '+', null);
+    '這週 以前'         => new Ast.Value.Date(Ast.DateEdge('end_of', 'week'), '+', null);
+    '這 禮拜 以前' => new Ast.Value.Date(Ast.DateEdge('end_of', 'week'), '+', null);
+    '上 星期'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '-', Ast.Value.Measure(1, 'week'));
+    '上週'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '-', Ast.Value.Measure(1, 'week'));
+    '上 禮拜'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'week'), '-', Ast.Value.Measure(1, 'week'));
+    '這個 月'           => new Ast.Value.Date(Ast.DateEdge('start_of', 'mon'), '+', null);
+    '今年'             => new Ast.Value.Date(Ast.DateEdge('start_of', 'year'), '+', null);
+    '下個 月'           => new Ast.Value.Date(Ast.DateEdge('end_of', 'mon'), '+', null);
+    '明年'             => new Ast.Value.Date(Ast.DateEdge('end_of', 'year'), '+', null);
+    '上個 月'           => new Ast.Value.Date(Ast.DateEdge('end_of', 'mon'), '-', Ast.Value.Measure(1, 'mon'));
+    '去年'             => new Ast.Value.Date(Ast.DateEdge('end_of', 'year'), '-', Ast.Value.Measure(1, 'year'));
+    }
+}
+
+constant_Date = {
+    !turking {
+    duration:constant_Measure_ms '以內' => new Ast.Value.Date(null, '+', duration);
+    duration:constant_Measure_ms '之內' => new Ast.Value.Date(null, '+', duration);
+    duration:constant_Measure_ms '以前' => new Ast.Value.Date(null, '-', duration);
+    duration:constant_Measure_ms '之前' => new Ast.Value.Date(null, '-', duration);
+    duration:constant_Measure_ms pt:constant_date_point '以後' => new Ast.Value.Date(pt.value, '+', duration);
+    duration:constant_Measure_ms pt:constant_date_point '之後' => new Ast.Value.Date(pt.value, '+', duration);
+    duration:constant_Measure_ms pt:constant_date_point '以前' => new Ast.Value.Date(pt.value, '-', duration);
+    duration:constant_Measure_ms pt:constant_date_point '之前' => new Ast.Value.Date(pt.value, '-', duration);
```

Comment:
I noticed that you're relying on the default definition of `constant_Measure_ms`. In English, that maps to a single `DURATION` token, because all the heavy lifting of understanding durations and time spans is done by CoreNLP through almond-tokenizer.
According to <https://corenlp.run>, this does not seem to be happening for Chinese. 
It will be good to understand what's going on, and what named entities we should rely on almond-tokenizer to understand, and what we'll let the neural parser learn.

(For English, NER is applied to locations, durations, times, dates and numbers, as well as a couple of language-agnostic regular expression patterns such as URLs, emails and hashtags; there is also an Americentric phone number recognizer that might need adaptation)

Reply:
Thanks for the explanation!
I knew this problem (kph, km/h ...) when translate constant, but have no good idea then though, planned to work on this today.

BTW I push to this branch every time I have some updates (not finished yet) in case there are too many stuff to review. Is this OK for you?

Reply:
Yeah that works, just tell me when you're ready to review. Although I went through it a couple times already so I don't expect there will be much left to comment on.

Reply:
Usually we say '時速 x 公里', meaning x km/h. This is what we use for measuring speed, most of the time we only use this (almost all the cases).

Reply:
There is no duration for Chinese in corenlp. 
I think we can use constant_date_point for concept like last week, next month, two weeks ago, after one year ... etc. As for duration, corenlp does not support now.

Reply:
Assuming durations in Chinese are simpler than English (where you have weird stuff like a "fortnight") you can also add duration support by defining
```
constant_Measure_ms = {
  num:constant_Number "unit 1" => addUnit(num, '...');
  num:constant_Number "unit 2" => addUnit(num, '...');
  ...
```
like we for other non-time units.